x-pack/plugin/enrich/src/main/java/org/elasticsearch/xpack/enrich/EnrichPolicyExecutor.java

```diff
@@ -49,10 +53,20 @@ private void tryLockingPolicy(String policyName) {
             throw new EsRejectedExecutionException("Policy execution failed. Policy execution for [" + policyName +
                 "] is already in progress.");
         }
+        if (policyExecutionPermits.tryAcquire() == false) {
```

Comment:
should we [block for a bit](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/Semaphore.html#tryAcquire-long-java.util.concurrent.TimeUnit-) here waiting on permit ? maybe 5 minutes ?

Reply:
I prefer to fail quickly here. If there are more than 50 policy executions then I guess usage of this api needs to be checked.